### PR TITLE
Task24/setup tests

### PR DESCRIPTION
**Pull request by rudyeila**
Setup the flask client for unit testing and added an example test. The same concept can be used for further tests. 

All tests must be within the tests/ folder. Could be a subfolder too. Folder name must start with test_*. The name of the test file must also start with test, i.e. test_*.py -> the name of each test function must also start with test: e.g. 
`def test_example(client):
     assert ...`
Pytest automatically detects all functions whose name starts with test_ and are inside the tests/ folder. The client argument in this case refers to the 'client' fixture that I defined in conftest.py... Pytest look for these function arguments that have the same name as the fixture and injects the fixture at runtime. Refer to the comments in conftest.py or to the pytest docs.

### CODE REVIEW

File: server/.gitignore [hunk 1]
```diff
@@ -143,3 +143,5 @@ dmypy.json
 .prof
```
Comment (NaseemSrour) [inline server/.gitignore:143]: We shouldn't contain local Git files in the repository. For example this file will affect my working directory when I pull the latest code to my computer.
Probably should remove it.

**Review by NaseemSrour — COMMENTED** [review]
Looks a good way to test Flask's app, the routes and their responses, easy and fast.
As I've read in Python's UnitTest documentation, this 3rd party lib 'PyTest' is great for Flask, meaning for the server routing. And the official 'UnitTest' Python lib should be used for unit tests in other sections.
With that in mind, I think we shouldn't close the issue this PR is appointed to. Let's see if we can create a new issue "Create Test for Server (Flask)" and appoint this PR to it first, before merging it. This way I can write the Unit Test code for issue #24, and keep the merging history clean.
What do you think?

File: server/.gitignore (hunk 1 above)
Comment (rudyeila) [reply server/.gitignore:143]: This is the idea from this .gitignore. It makes sure local files that are unrelated to this code are not commited and are automatically ignored.

**Comment by rudyeila** [comment]
Pytest is simply a third party testing library. It should be "better" than the default unittesting library from python. It is also a unit testing library and can be used to unit test all sort of code. It is a matter of taste. 

To test the server, we need to have some functionality implemented already, which we don't yet. I would suggest we merge this code, close the task, and then start working on implementing the functionality for the server, and writing tests at the same time. 

So for example, you add the functionality to handle POSTs to /item, you should also write the unit tests for that functionality. All in the same PR. I think this is the best way to do it.

**Comment by NaseemSrour** [comment]
Oh I see what you mean. Yeah that sounds about right.
Ok let's close this and continue as you suggested.

**Comment by rudyeila** [comment]
Yes it should work. Any unit test you should be able to write in the default library should be also doable in pytest.

File: server/.gitignore (hunk 1 above)
Comment (NaseemSrour) [reply server/.gitignore:143]: I was thinking this gitignore file will contain future local log files (paths to them), and the paths in this gitignore file will not align with the logs' paths in others' working dir. See what I mean?

**Review by NaseemSrour — APPROVED** [review]

File: server/.gitignore (hunk 1 above)
Comment (rudyeila) [reply server/.gitignore:143]: That's why all the paths are relative. So they should align for everyone. If I understand what you mean at least.

**Pull request merged**